src/backend/distributed/utils/maintenanced.c

```diff
@@ -246,6 +246,17 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 	/* make worker recognizable in pg_stat_activity */
 	pgstat_report_appname("Citus Maintenance Daemon");
 
+	/* before doing the actual jobs, check any version mismatches */
+	StartTransactionCommand();
+
+	if (!CheckCitusVersion(LOG))
+	{
+		/* return code of 1 requests worker restart */
+		proc_exit(1);
```

Comment:
An alternative would be to not start the daemon unless `CheckCitusVersion` by splitting up `StartupCitusBackend()`.